systemd-pcrlock introduced in 2.1.7 cannot build with libsystemd <258

**Describe the bug**

Trying to compile with libsystemd <258 (in this instance, 257) fails with:

```
FAILED: [code=1] plugins/systemd-pcrlock/libfu_plugin_systemd_pcrlock.a.p/fu-systemd-pcrlock-plugin.c.o 
cc -Iplugins/systemd-pcrlock/libfu_plugin_systemd_pcrlock.a.p -Iplugins/systemd-pcrlock -I../plugins/systemd-pcrlock -I. -I.. -Ilibfwupd -I../libfwupd -Ilibfwupdplugin -I../libfwupdplugin -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/p11-kit-1 -I/usr/include/libusb-1.0 -I/usr/include/libxmlb-2 -flto=8 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c17 -Wfatal-errors -Waggregate-return -Wunused -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wdiscarded-qualifiers -Wduplicated-branches -Wduplicated-cond -Wempty-body -Wfloat-equal -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Wimplicit-int -Wincompatible-pointer-types -Winit-self -Wint-conversion -Wlogical-op -Wmaybe-uninitialized -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-cast-function-type -Wno-deprecated-declarations -Wno-address-of-packed-member -Wno-unknown-pragmas -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wno-nonnull-compare -Wno-analyzer-use-of-uninitialized-value -Wno-analyzer-fd-double-close -Wold-style-definition -Woverride-init -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wunused-variable -Wvla -Wwrite-strings -fstack-protector-strong -DFWUPD_COMPILATION -D_DEFAULT_SOURCE -D_BSD_SOURCE -D__BSD_VISIBLE -D_GNU_SOURCE -D_DARWIN_C_SOURCE=900000 -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -pthread -isystem/usr/include/glib-2.0 -isystem/usr/lib/x86_64-linux-gnu/glib-2.0/include '-DG_LOG_DOMAIN="FuPluginSystemdPcrlock"' -MD -MQ plugins/systemd-pcrlock/libfu_plugin_systemd_pcrlock.a.p/fu-systemd-pcrlock-plugin.c.o -MF plugins/systemd-pcrlock/libfu_plugin_systemd_pcrlock.a.p/fu-systemd-pcrlock-plugin.c.o.d -o plugins/systemd-pcrlock/libfu_plugin_systemd_pcrlock.a.p/fu-systemd-pcrlock-plugin.c.o -c ../plugins/systemd-pcrlock/fu-systemd-pcrlock-plugin.c
In file included from /usr/lib/x86_64-linux-gnu/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:34,
                 from /usr/include/glib-2.0/glib/galloca.h:34,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:24,
                 from ../libfwupd/fwupd-bios-setting.h:9,
                 from ../libfwupd/fwupd.h:11,
                 from ../libfwupdplugin/fwupdplugin.h:11,
                 from ../plugins/systemd-pcrlock/fu-systemd-pcrlock-plugin.h:9,
                 from ../plugins/systemd-pcrlock/fu-systemd-pcrlock-plugin.c:13:
../plugins/systemd-pcrlock/fu-systemd-pcrlock-plugin.c: In function ‘glib_autoptr_clear_sd_varlink_interface’:
../plugins/systemd-pcrlock/fu-systemd-pcrlock-plugin.c:23:53: error: ‘sd_varlink_interface_free’ undeclared (first use in this function); did you mean ‘sd_varlink_interface’?
   23 | G_DEFINE_AUTOPTR_CLEANUP_FUNC(sd_varlink_interface, sd_varlink_interface_free)
      |                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~
```

`sd_varlink_interface_free()` was introduced in libsystemd 258, but meson isn't doing any version constraints

**Steps to Reproduce**

Try to compile the latest 2.1.7 tag on a system with libsystemd <258, for example the freedesktop-sdk 25.08 toolchain:

https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/33494

**Expected behavior**

Older libsystemd's are supported by the plugn, or constraints are added to explicitly enforce checking . Could the cleanup code be made conditional, or the plugin itself made optional and not bound to the `systemd` meson option?

**fwupd version information**

2.1.7

--- a/plugins/systemd-pcrlock/meson.build
+++ b/plugins/systemd-pcrlock/meson.build
@@ -1,5 +1,6 @@
 host_machine.system() == 'linux' or subdir_done()
 libsystemd.found() or subdir_done()
+libsystemd.version().version_compare('>= 258') or subdir_done()
 cc.has_header('systemd/sd-varlink.h', dependencies: libsystemd) or subdir_done()
 
 cargs = ['-DG_LOG_DOMAIN="FuPluginSystemdPcrlock"']

